File: Dictionaries and Sets/128. Longest Consecutive Sequence/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        sett = set(nums)
        longest = 0

        for n in sett:
            if n - 1 not in sett:
                curr = n
                curr_length = 1
                while curr + 1 in sett:
                    curr += 1
                    curr_length += 1

                longest = max(longest, curr_length)

        return longest 

# Time: O(n)
# Space: O(n)

File: Dictionaries and Sets/128. Longest Consecutive Sequence/128. Longest Consecutive Sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  const sett = new Set(nums);
  let longest = 0;

  for (let n of sett) {
    if (!sett.has(n - 1)) {
      let curr = n;
      let curr_length = 1;
      while (sett.has(curr + 1)) {
        curr++;
        curr_length++;
      }
      longest = Math.max(longest, curr_length);
    }
  }

  return longest;
};

// Time: O(n)
// Space: O(n)
